### PR TITLE
fix: Look for sentry.properties on flavor folder

### LINKED ISSUE #716: Gradle Plugin Not handling flavors/dimensions properly
When you have multiple flavors/dimensions in the project, the sentry.properties file does not work when it is placed in a folder with a combination of flavors from different dimensions.

CC: @justacodefan

### PR DESCRIPTION
**Pull request by bruno-garcia**
/cc @ninniuz

Resolves #716

### CODE REVIEW

File: sentry-android-gradle-plugin/src/main/groovy/io/sentry/android/gradle/SentryPlugin.groovy [hunk 1]
```diff
@@ -204,23 +204,36 @@ class SentryPlugin implements Plugin<Project> {
...
+                        variant.productFlavors.each {
+                            // flavors used with dimension come in second
+                            possibleProps.push("${project.projectDir}/src/${it.name}/${propName}")
+                        }
+
+                        possibleProps = [
                                 "${project.projectDir}/src/${flavorName}/${propName}",
+                                "${project.rootDir.toPath()}/src/${flavorName}/${propName}",
+                        ] + possibleProps + [
+                                "${project.projectDir}/src/${variantName}/${propName}",
```
Comment (ninniuz) [inline sentry-android-gradle-plugin/src/main/groovy/io/sentry/android/gradle/SentryPlugin.groovy]: I think the `variantName` stuff should go before the product flavors.

File: sentry-android-gradle-plugin/src/main/groovy/io/sentry/android/gradle/SentryPlugin.groovy [hunk 2]
```diff
@@ -204,23 +204,36 @@ class SentryPlugin implements Plugin<Project> {
...
+                        // current flavor name takes priority
+                        def possibleProps = []
+                        variant.productFlavors.each {
+                            // flavors used with dimension come in second
+                            possibleProps.push("${project.projectDir}/src/${it.name}/${propName}")
+                        }
+
+                        possibleProps = [
                                 "${project.projectDir}/src/${flavorName}/${propName}",
+                                "${project.rootDir.toPath()}/src/${flavorName}/${propName}",
```
Comment (ninniuz) [inline sentry-android-gradle-plugin/src/main/groovy/io/sentry/android/gradle/SentryPlugin.groovy]: Is there any reason why we are searching in `${project.rootDir.toPath()}/src/`? 
I think I have never seen an Android project with a `src` folder in its root 🤔

File: sentry-android-gradle-plugin/src/main/groovy/io/sentry/android/gradle/SentryPlugin.groovy (hunk 1 above)
Comment (bruno-garcia) [reply sentry-android-gradle-plugin/src/main/groovy/io/sentry/android/gradle/SentryPlugin.groovy]: fair. Didn't want to rename it either in the same PR but that's actually the `buildType` as you know

File: sentry-android-gradle-plugin/src/main/groovy/io/sentry/android/gradle/SentryPlugin.groovy (hunk 2 above)
Comment (bruno-garcia) [reply sentry-android-gradle-plugin/src/main/groovy/io/sentry/android/gradle/SentryPlugin.groovy]: I wanted to get rid of this but that'd could *possibly* be a breaking change. So I'd say we leave this for now and take into account on `2.0.0`.

**Pull request merged**